src/weather.py

```diff
@@ -29,24 +28,69 @@ def fetch_weather(self):
         }
         
         response = requests.get(base_url, params=params)
-        
+
         if response.status_code == 200:
             data = response.json()
             self.temperature = data['main']['temp']
             self.humidity = data['main']['humidity']
             self.weather_description = data['weather'][0]['description']
             self.wind_speed = data['wind']['speed']
         else:
+            print("We failed to fetch weather data")
             raise Exception(f"Error fetching weather data: {response.status_code} - {response.json().get('message')}")
 
     def display_weather(self):
-        print(f"Weather in {self.city}:")
+        # Try to print out the prettier version if we have it.
+        print(f"Weather in {self.pretty_name if self.pretty_name else self.city}:")
         print(f"  Description: {self.weather_description}")
         print(f"  Temperature: {self.temperature}°F")
         print(f"  Humidity: {self.humidity}%")
         print(f"  Wind Speed: {self.wind_speed} m/s")
 
 
+def get_owm_query(city_input: str) -> str:
```

Comment:
All of this could get cleaned up if I was able to use a singular tool for city search and weather lookup, like pyowm. Will look into this if I have some extra time beyond reason.